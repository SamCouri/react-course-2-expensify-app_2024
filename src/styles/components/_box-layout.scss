.box-layout {
    align-items: center;
    background: url("/images/bg.jpg");
    background-size: cover;
    display: flex;
    height: 100vh; //viewport hight
    justify-content: center;
    width: 100vw; //viewport width
}

.box-layout__box {
    background: fade-out(white, 0.15);
    border-radius: 3px;
    padding: $l-size $m-size;  // top and bottom, left and right
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    margin: 0 0 $m-size 0; //top, right, bottom, left
    line-height: 1;
}