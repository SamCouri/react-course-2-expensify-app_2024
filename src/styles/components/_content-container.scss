.content-container {
    margin: 0 auto;
    padding: 0 $m-size; //(top and bottom 0) left and right $m-size
    max-width: 80rem;
}