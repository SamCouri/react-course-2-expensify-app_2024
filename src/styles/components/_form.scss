.form {
    display: flex;
    flex-direction: column;  
    > * {    //this is to target all Form direct children (description, amount date, etc.)
        margin-bottom: $m-size; 
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}