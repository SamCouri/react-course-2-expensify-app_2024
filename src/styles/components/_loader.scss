.loader {
    align-items: center;  //center vertically
    display: flex;
    height: 100vh; //viewport hight
    justify-content: center; //center horizontally
    width: 100vw; //viewport width
}

.loader__image {
    height: 6rem;
    width: 6rem;
}