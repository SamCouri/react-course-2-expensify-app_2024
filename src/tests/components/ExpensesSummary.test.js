import React from "react";
import { shallow } from "enzyme";
import  { ExpensesSummary } from "../../components/ExpensesSummary";

test("should correctly render ExpenseSummary with 1 expense totaling $94.34", ()=> {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
    })


test("should correctly render ExpenseSummary multiple expenses totaling $94.34", ()=> {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={94.34}/>);
    expect(wrapper).toMatchSnapshot();
    })

