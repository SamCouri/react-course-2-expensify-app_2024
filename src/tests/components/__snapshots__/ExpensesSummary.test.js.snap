// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary multiple expenses totaling $94.34 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totaling 
    $0.94
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with 1 expense totaling $94.34 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;
